libs/category/src/filters/behaviors/apply/apply-filter.spec.ts

```diff
@@ -1,17 +1,154 @@
-describe('daffApplyFilter', () => {
-  it('should apply a toggle request', () => {
+import { DaffCategoryFilterEqualFactory, DaffCategoryFilterRangeNumericFactory, DaffCategoryFilterRequestEqualFactory, DaffCategoryFilterRequestRangeNumericFactory } from "@daffodil/category/testing";
+import { DaffCategoryEqualFilter, DaffCategoryFilter, DaffCategoryFilterEqualRequest, DaffCategoryFilterRangeRequest, DaffCategoryFilterType } from "../../../models/public_api";
+import { DaffCategoryFilterRequestNameMismatch } from "../../errors/request-name-mismatch.error";
+import { DaffCategoryFilterRequestTypeMismatch } from "../../errors/request-type-mismatch.error";
+import { DaffCategoryUnknownFilterType } from "../../errors/unknown-filter-type.error";
+import { daffApplyFilter } from "./apply-filter";
 
+describe('@daffodil/category | filters | behaviors | apply | daffApplyFilter', () => {
+  it('should apply an equal filter request', () => {
+		const filter: DaffCategoryEqualFilter = new DaffCategoryFilterEqualFactory().create({
```

Comment:
Can you DI the factories over direct instantiation? I'm kinda torn here because these are pure functions, but that feels more maintainable.

Reply:
Weren't you talking before about how you'd prefer direct instantiation over DI with factories?

Reply:
@lderrickable have a ref to that earlier discussion? I very well may have, but looking at this in light of the solution's I've applied in the branch to resolve:

https://github.com/graycoreio/daffodil/issues/882

and realising how we may finally be able to resolve:

Reply:
@damienwebdev I think what @lderrickable is referring to here is when we were talking about directly instantiating SUTs, for example newing up an effects class in the effects test rather than DIing it.

Reply:
I can't remember when exactly I thought you said that, and I definitely could be mistaken.

Do you want to DI the factories so there are fewer instantiations throughout the tests? In this case with pure functions would there be a difference between:
```
let factory: DaffFactory;

beforeEach(() => {
   TestBed.configureTestingModule({
      providers: [
        DaffFactory
      ],
   });

   factory = TestBed.inject(DaffFactory);
});
```
and
```
let factory: DaffFactory;

beforeEach(() => {
   factory = new DaffFactory();
});
```
?
I like the second one more, but it isn't DI.

Reply:
You shouldn't need to provide the factory

Reply:
In theory, it should be as simple as:

```ts
let factory: DaffFactory;

beforeEach(() => {
   TestBed.configureTestingModule();

   factory = TestBed.inject(DaffFactory);
});
```

The power of the di (for the factories) comes from

```ts
let factory: DaffFactory;

beforeEach(() => {
   TestBed.configureTestingModule();

   factory = TestBed.inject(DaffSomeFactoryWithLike5DifferentDepsThatHaveTheirOwnDeps);
});
```

Reply:
updated